### DISCUSSION TITLE
Searching within a specific folder leads to an invalid regex error

### DISCUSSION THREAD

**Discussion by asastry-jumptrading** [community/community#88721] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Search in this repo: https://github.com/gcc-mirror/gcc

for the text `ordered_set`. On the left hand pane, I see that I can filter the search to specific folders, including `libstdc++-v3/`. Then I click on that folder. 

Expected: Only displays results within that folder

Actual: No results displayed, this error:
```
Invalid regular expression
repo:gcc-mirror/gcc ordered_set path:/^libstdc++-v3\//
```

I'm guessing the `+` or `-` need to be escaped?

**Comment by AlessioBugetti** [comment]
I believe you need to escape the special characters in your search query. You can try: `repo:gcc-mirror/gcc ordered_set path:/^libstdc\+\+-v3\//`.

**Reply by asastry-jumptrading** [reply]
I'm not typing the search query, this is generated by the GitHub UI when I click the folder name in the left hand side pane. GitHub should be escaping those characters when generating the query.
